ark/phenotyping/create_cell_som.R

```diff
@@ -16,6 +16,15 @@ library(arrow)
 library(data.table)
 library(FlowSOM)
 
+# a helper function for computing 99.9%
+percentile_99_9_helper <- function(x) {
+    if (quantile(as.numeric(x[x > 0]), 0.999) == 0) {
+        return(quantile(as.numeric(x[x > 0], 0.999), 1))
```

Comment:
I'm confused by this line. It seems like you are turning `x[x>0],0.999` into a numeric (parentheses seem to be in the wrong place)? If 99.9% is 0, should just return the max. So you can use either `return(quantile(x,1))` or `return(max(x))`. No need to have 0.999 in this return line.

Reply:
Thanks for pointing that out, I forgot to commit that change.

Reply:
I need to convert to `numeric` because `percentile_99_9_helper` reads `x` as a character type by default even if the values in `clusterCountsNorm` are numeric.

Reply:
Right, that part is fine. It's just that the parentheses for the as.numeric are around the 0.999 right now.